server/pbench/bin/index-pbench

```diff
@@ -2073,25 +2074,24 @@ def main(options, args):
 
 ###########################################################################
 # Options handling
-
-prog_options = [
-    make_option("-C", "--config", dest="cfg_name", help="Specify config file"),
-    make_option("-E", "--indexing-errors", dest="indexing_errors", help="Name of a file to write JSON documents that fail to index"),
-    make_option("-D", "--temp-directory", dest="tmpdir", help="Temporary directory to use while indexing"),
-    make_option("-M", "--metadata", dest="metadata_string", help="Specify additional metadata (e.g. for browbeat) as a JSON document string"),
-    # options for debuggging and unit testing
-    make_option("-U", "--unittest", action="store_true", dest="debug_unittest", help="Run in unittest mode"),
-    make_option("-T", "--time-ops", action="store_true", dest="debug_time_operations", help="Time action making routines"),
-]
-
 if __name__ == '__main__':
-    parser = OptionParser("Usage: index-pbench [--config <path-to-config-file>] [--metadata \"<JSON doc>\"] [--working-directory <dir>] <path-to-tarball>")
-    for o in prog_options:
-        parser.add_option(o)
+    parser = ArgumentParser("Usage: index-pbench [--config <path-to-config-file>] [--metadata \"<JSON doc>\"] [--working-directory <dir>] <path-to-tarball>")
+    parser.add_argument("-C", "--config", dest="cfg_name", help="Specify config file")
+    parser.add_argument("-E", "--indexing-errors", dest="indexing_errors", help="Name of a file to write JSON documents that fail to index")
+    parser.add_argument("-D", "--temp-directory", dest="tmpdir", help="Temporary directory to use while indexing")
+    parser.add_argument("-M", "--metadata", dest="metadata_string", help="Specify additional metadata (e.g. for browbeat) as a JSON document string")
+    # options for debuggging and unit testing
+    parser.add_argument("-U", "--unittest", action="store_true", dest="debug_unittest", help="Run in unittest mode")
+    parser.add_argument("-T", "--time-ops", action="store_true", dest="debug_time_operations", help="Time action making routines")
+    # argparse also handles the cli argument passed to the script, therefore this parser has been added
+    parser.add_argument("path_to_tarball_file",nargs=1,help="specify the path to the tar.gz file")
     parser.set_defaults(cfg_name = os.environ.get("ES_CONFIG_PATH"))
     parser.set_defaults(tmpdir = os.environ.get("TMPDIR"))
-
-    (options, args) = parser.parse_args()
-
+    parsed = parser.parse_args()
+    #parse_args returns a namespace object. Therefore this is a minimal serialisation that has been added to process namespaces to normal dictionary
+    parsed = vars(parsed)
+    #argparse also handles the cli argument. Therefore the dictionary already had that element. To make the output argparse similar to optparse this element has been popped
+    args = parsed.pop("path_to_tarball_file")
+    options = parsed
```

Comment:
The unit tests failures are caused by the following: `options` was a `Values` object in optparse, not a dict, so you could refer to an attribute as `options.debug_time_operations`. Now that it is a dict, you have to use index notation: `options['debug_time_operations']` (remember, this is Python, not Javascript ;-) ). But why convert it to a dict in the first place? You could leave it as a Namespace and refer to `parsed.debug_time_operations`.